schemas/advanced-layouts.md

```diff
@@ -118,3 +118,28 @@ If we were to take an IPLD node, and assert that it is of type `BigConfidentialC
 5. The `data` property of the `map` is of `map` kind, which itself should conform to the `HashMapRoot` type specification, but must be interacted through with the logic associated with `HashMap` in order to make sense of it (which may also involve loading further blocks to traverse the sharded data).
 
 If `ROT13` and `HashMap` were to omit their `root` descriptor, we could only make assertions 1 and 2 above.
+
+## Representation parameterization
+
+The IPLD HashMap spec defines some possible parameters that may be used at instantiation time. These parameters are stored in the root block so the parameters used to write the `HashMap` are known when reading the `HashMap`. Such parameters are generally provided programatically, but we may use our advanced layout definition to dictate strict requirements.
+
+```ipldsch
+advanced HashMap {
+  root HashMapRoot
+  parameter hashAlg String
```

Comment:
I'm tempted to suggest an "optional" on the end here too, I can see use cases for these things being always required by a schema reference and sometimes being left up to other layers (programatic) or allowing for flexibility in the data that's loaded.